packages/quantic/force-app/main/default/lwc/quanticRecentResultsList/__tests__/quanticRecentResultList.test.js

```diff
@@ -0,0 +1,496 @@
+  QuanticRecentResultsList,
+  'c-quantic-recent-results-list',
+  defaultOptions
+);
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildRecentResultsList: functionMocks.buildRecentResultsList,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticRecentResultsList && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticRecentResultsList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+describe('c-quantic-recent-result-list', () => {
+  afterEach(() => {
+    cleanup();
+    recentResultsListState = initialRecentResultsListState;
+    isInitialized = false;
+  });
+
+  describe('controller initialization', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    it('should display the placeholder component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const placeholder = element.shadowRoot.querySelector(
+        selectors.placeholder
+      );
+
+      expect(placeholder).not.toBeNull();
+      expect(
+        element.shadowRoot.querySelector(selectors.quanticCardContainer)
+      ).toBeNull();
+    });
+
+    it('should build the controller and subscribe to its state', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.buildRecentResultsList).toHaveBeenCalledTimes(1);
+      expect(functionMocks.buildRecentResultsList).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          initialState: {results: defaultResults},
+          options: expect.objectContaining({
+            maxLength: defaultMaxLength,
+          }),
+        }
+      );
+      expect(functionMocks.recentResultsListSubscriber).toHaveBeenCalledTimes(
+        1
+      );
+    });
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    it('should display the error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when the component is initialized', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+      jest
+        .spyOn(mockHeadlessLoader, 'getHeadlessEnginePromise')
+        .mockImplementation(() => Promise.resolve(exampleEngine));
```

Comment:
I don't really understand why you're doing this instead of just mocking `headlessLoader` like in all the other test files?

Reply:
Ah okay, other tests simply mock that at the top of the test file instead of in a `beforeEach` of just some tests.